src/deepforest/visualize.py

```diff
@@ -10,27 +10,105 @@
 import random
 import warnings
 import supervision as sv
+from typing import Optional, Union
 from deepforest.utilities import determine_geometry_type
 
 
+def _load_image(image: Optional[Union[np.typing.NDArray, str, Image.Image]] = None,
+                df: Optional[pd.DataFrame] = None,
+                root_dir: Optional[str] = None) -> np.typing.NDArray:
+    """Utility function to load an image from either a path or a
+    prediction/annotation dataframe.
+
+    Returns an image in RGB format with HWC channel ordering.
+
+    Args:
+        image (optional): Numpy array or string
+        df (optiona): Pandas dataframe
+
+    Returns:
+        image: Numpy array
+    """
+
+    if image is None and df is None:
+        raise ValueError(
+            "Either an image or a valid dataframe must be provided for plotting.")
+    elif df is not None:
+        if not root_dir and not hasattr(df, 'root_dir'):
+            raise ValueError(
+                "The 'root_dir' attribute does not exist in the dataframe. Please specify the 'root_dir' argument."
+            )
+        else:
+            root_dir = df.root_dir
+            # expected str, bytes or os.PathLike object, not Series
+            root_dir = df.root_dir.iloc[0] if isinstance(root_dir,
+                                                         pd.Series) else root_dir
+
+        image_path = os.path.join(root_dir, df.image_path.unique()[0])
+        image = np.array(Image.open(image_path))
+    elif isinstance(image, str):
+        image = np.array(Image.open(image))
+    elif isinstance(image, Image.Image):
+        image = np.array(image)
+    elif not isinstance(image, np.ndarray):
+        raise ValueError("Image")
+
+    # Fix channel ordering
+    if image.shape[0] == 3:
+        warnings.warn("Input images must be channels last format [h, w, 3] not channels "
+                      "first [3, h, w], using np.rollaxis(image, 0, 3) to invert!")
+        image = np.rollaxis(image, 0, 3)
```

Comment:
```suggestion
        image = np.transpose(image, (1, 2, 0))
```
Could this be more clear ? not really important